roles/common/templates/logrotate_rules.j2

```diff
@@ -0,0 +1,120 @@
+# {{ ansible_managed | comment }}
+# Logrotate configuration for {{ item.name }}
+
+{% for path in item.paths %}
+{{ path }}
+{% endfor %} {
+{% if item.options.daily is defined and item.options.daily %}
+    daily
+{% elif item.options.weekly is defined and item.options.weekly %}
+    weekly
+{% elif item.options.monthly is defined and item.options.monthly %}
+    monthly
+{% elif item.options.yearly is defined and item.options.yearly %}
+    yearly
+{% endif %}
+{% if item.options.rotate is defined %}
+    rotate {{ item.options.rotate }}
+{% endif %}
+{% if item.options.size is defined %}
+    size {{ item.options.size }}
+{% endif %}
+{% if item.options.maxage is defined %}
+    maxage {{ item.options.maxage }}
+{% endif %}
+{% if item.options.compress is defined and item.options.compress %}
+    compress
+{% endif %}
+{% if item.options.nocompress is defined and item.options.nocompress %}
+    nocompress
+{% endif %}
+{% if item.options.delaycompress is defined and item.options.delaycompress %}
+    delaycompress
+{% endif %}
+{% if item.options.missingok is defined and item.options.missingok %}
+    missingok
+{% endif %}
+{% if item.options.nomissingok is defined and item.options.nomissingok %}
+    nomissingok
+{% endif %}
+{% if item.options.notifempty is defined and item.options.notifempty %}
+    notifempty
+{% endif %}
+{% if item.options.ifempty is defined and item.options.ifempty %}
+    ifempty
+{% endif %}
+{% if item.options.copytruncate is defined and item.options.copytruncate %}
+    copytruncate
+{% endif %}
+{% if item.options.nocopytruncate is defined and item.options.nocopytruncate %}
+    nocopytruncate
+{% endif %}
+{% if item.options.create is defined and item.options.create %}
+    create {{ item.options.create_mode | default('0644') }} {{ item.options.create_owner | default('root') }} {{ item.options.create_group | default('root') }}
+{% endif %}
+{% if item.options.nocreate is defined and item.options.nocreate %}
+    nocreate
+{% endif %}
+{% if item.options.sharedscripts is defined and item.options.sharedscripts %}
+    sharedscripts
+{% endif %}
+{% if item.options.nosharedscripts is defined and item.options.nosharedscripts %}
+    nosharedscripts
+{% endif %}
+{% if item.options.dateext is defined and item.options.dateext %}
+    dateext
+{% endif %}
+{% if item.options.nodateext is defined and item.options.nodateext %}
+    nodateext
+{% endif %}
+{% if item.options.dateformat is defined %}
+    dateformat {{ item.options.dateformat }}
+{% endif %}
+{% if item.options.extension is defined %}
+    extension {{ item.options.extension }}
+{% endif %}
```

Comment:
Same thing with all of this stuff - we should define a standard format that we want all our rotated logs to use. Otherwise we will be back in the position we've been in, where some logs do one thing and others do something different.

Reply:
but this is how logging works. If for example we define for the median... then suddenly our nginx load balancer is reloading loads more because we've decided this is our tolerance level to balance it out with a rails app that is less taxed. Or even the same nginx (passenger) configs will defer on a application web server vs a load balancer.